src/ui/surfaces/SiteConfiguration.tsx

```diff
@@ -0,0 +1,232 @@
+import {
+  Button,
+  Card,
+  CardLoader,
+  CardTitle,
+  Checkbox,
+  Form,
+  FormField,
+  SiteAccessConfigurationSurface,
+} from "@netlify/sdk/ui/react/components";
+import { trpc } from "../trpc";
+import { useNetlifySDK } from "@netlify/sdk/ui/react";
+import { useEffect, useState } from "react";
+import { z } from "zod";
+
+export const cspConfigFormSchema = z.object({
+  reportOnly: z.boolean().optional(),
+  reportUri: z.string().url().optional(),
+  unsafeEval: z.boolean().optional(),
+  path: z.string(),
+  excludedPath: z.string().optional(),
+});
+
+export const SiteConfiguration = () => {
+  const [triggerTestRun, setTriggerTestRun] = useState(false);
+  const sdk = useNetlifySDK();
+  const trpcUtils = trpc.useUtils();
+  const siteConfigQuery = trpc.siteConfig.queryConfig.useQuery();
+  const siteConfigurationMutation = trpc.siteConfig.mutateConfig.useMutation({
+    onSuccess: async () => {
+      await trpcUtils.siteConfig.queryConfig.invalidate();
+    },
+  });
+  const siteEnablementMutation = trpc.siteConfig.mutateEnablement.useMutation({
+    onSuccess: async () => {
+      await trpcUtils.siteConfig.queryConfig.invalidate();
+    },
+  });
+  const siteDisablementMutation = trpc.siteConfig.mutateDisablement.useMutation(
+    {
+      onSuccess: async () => {
+        await trpcUtils.siteConfig.queryConfig.invalidate();
+      },
+    }
+  );
+  const triggerConfigTestMutation =
+    trpc.siteConfig.mutateTriggerConfigTest.useMutation({
+      onSuccess: async () => {
+        await trpcUtils.siteConfig.queryConfig.invalidate();
+      },
+    });
+
+  const onEnableHandler = () => {
+    siteEnablementMutation.mutate();
+  };
+
+  const onDisableHandler = () => {
+    siteDisablementMutation.mutate();
+  };
+
+  useEffect(() => {
+    if (triggerTestRun) {
+      document
+        .getElementsByTagName("form")[0]
+        ?.dispatchEvent(new Event("submit", { bubbles: true }));
+
+      setTriggerTestRun(false);
+    }
+  }, [triggerTestRun]);
+
+  if (siteConfigQuery.isLoading) {
+    return <CardLoader />;
+  }
+
+  const onSubmitTest = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    // Triggers the submit of the form in useEffect
+    setTriggerTestRun(true);
+  };
+
+  type CspConfigFormData = z.infer<typeof cspConfigFormSchema>;
+
+  const onSubmit = ({
+    path: newPath,
+    excludedPath: newExcludedPath,
+    ...data
+  }: CspConfigFormData) => {
+    const path =
+      newPath === ""
+        ? []
+        : newPath.split("\n").filter((path) => path.trim() !== "");
+
+    const excludedPath =
+      !newExcludedPath || newExcludedPath === ""
+        ? []
+        : newExcludedPath.split("\n");
+
+    void (async () => {
```

Comment:
I think you can make onSubmit be async, because Form doesn't check the return type